### DISCUSSION TITLE
Question about customizing date picker month selection menu

### DISCUSSION THREAD

**Discussion by Ryandi-S** [rsuite/rsuite#2134] [Q&A]
Hi. I'm trying to use custom styling for month selection menu in date picker :
<img width="583" alt="Screen Shot 2021-11-16 at 13 04 22" src="https://user-images.githubusercontent.com/93305443/141930159-0a50b564-de8b-4e05-b2cf-7333f95c5f66.png">
but apparently the whole structure is generated by js, including the content updating while scrolling. This make it impossible for me to customize the appearance of the month selection menu. Because the `rs-calendar-month-dropdown-row` got inline style that is used for the js calculation. If I force change it with `!important` it will just break the whole component. Is there any way to customize this part? Thanks.

**Comment by simonguo** [comment]
@Ryandi-S  Is the following the month selector you want?
```jsx
<DateRangePicker format="yyyy-MM" />
<DatePicker format="yyyy-MM" />
```

![image](https://user-images.githubusercontent.com/1203827/143406732-2dee5808-36db-4b4c-a15e-b5e71bc1fccc.png)

**Comment by edward-meister** [comment]
we need a method to render custom element for this

**Comment by simonguo** [comment]
I added a `monthDropdownProps` prop for customizing the month selection menu.  
https://github.com/rsuite/rsuite/pull/3913

If you just want to add a `className` to the row, you can use `itemClassName`.

```tsx
<DatePicker
  monthDropdownProps={{
    itemClassName: 'custom-item'
  }}
/>
```

If you want to deeply customize the month selection menu, you can customize the component through `as` and `itemAs`.
```tsx
const Menu = props => {
  const { className, ...rest } = props;
  return <ul className={`${className} custom-menu`} {...rest} />;
};

const Item = props => {
  const { className, ...rest } = props;
  return <li className={`${className} custom-item`} {...rest} />;
};

render(
  <DatePicker
    monthDropdownProps={{
      as: Menu,
      itemAs: Item
    }}
  />
);
```
